components/script/dom/window.rs

```diff
@@ -1984,6 +1986,23 @@ impl Window {
             return;
         }
 
+        // Send resize message to any local `Pipeline`s synchronously, so that the value
+        // can be reflected in the next layout.
+        let device_pixel_ratio = self.device_pixel_ratio();
+        with_script_thread(|script_thread| {
+            for iframe_size in new_iframe_sizes.values() {
+                script_thread.handle_resize_message(
+                    iframe_size.pipeline_id,
+                    WindowSizeData {
+                        initial_viewport: iframe_size.size,
+                        device_pixel_ratio,
+                    },
+                    // TODO: This might send an extra resize event. This can fixed by explicitly
+                    // supporting `<iframe>` creation when the size isn't known yet.
+                    WindowSizeType::Resize,
+                );
+            }
+        });
         // Send asynchronous updates to `Constellation.`
```

Comment:
I think all of these will now result in receiving resize messages back from the constellation, in which case we need to write down somewhere what the proposed re-design is, which probably should include making a distinction between resizes that have already been batched locally at the script-thread(for same-origin iframes), and those that need to be propagated to other script-threads via the constellation. Describing this in an issue for a follow-up is a good idea I think.

Reply:
Also one last note on the topic of "According to the
specification these frames should all be synchrnously laid out", I think the way to do that in Servo would be to still first batch the resizes locally, but somehow only process them as part of an rendering update when the cross-origin resizes have been taken into account as well. So in other words, instead of batching them for the next udpate, they'd have to be batched for "an update in the future" which would somehow take into account the resizing of the cross-origin iframes, and this "transaction" across script-thread would have be to brokered by the constellation. I think it's also a good idea to start discussing in more details in an issue.

Reply:
It's a great idea to document the new design. This change fixes the issue in the majority of cases, but there are still some tricky cases where many resize events flooding the `Constellation` could lead to flashing layout. I hope to address those in a followup change, but the solution will affect the final design. Maybe I can have a section of the book ready with the design once I've worked through all of the issues.

Regarding whether or not we need synchronous layout of the entire frame tree (I'm not convinced we don't need it yet), I agree it's good to discuss more in #34655.